mifosng-android/src/main/java/com/mifos/mifosxdroid/online/clientidentifiers/ClientIdentifiersFragment.java

```diff
@@ -130,16 +132,34 @@ public void showUserInterface() {
     public void showClientIdentifiers(List<Identifier> identifiers) {
         this.identifiers = identifiers;
         identifierListAdapter.setIdentifiers(identifiers);
+        identifierListAdapter.notifyDataSetChanged();
+
+        if (identifiers.isEmpty()) {
+            showEmptyClientIdentifier();
+        } else {
+            if (ll_error.getVisibility() == View.VISIBLE) {
+                ll_error.setVisibility(View.GONE);
+            }
+        }
     }
 
-    @Override
-    public void showEmptyClientIdentifier() {
+    private void showEmptyClientIdentifier() {
         ll_error.setVisibility(View.VISIBLE);
         mNoIdentifierText.setText(getResources().getString(R.string.no_identifier_to_show));
         mNoIdentifierIcon.setImageResource(R.drawable.ic_assignment_turned_in_black_24dp);
     }
 
     @Override
+    public void onClientIdentifierCreationSuccess() {
+        loadIdentifiers();
+    }
+
+    @Override
+    public void onClientIdentifierCreationFailure() {
+
```

Comment:
Sorry! It was handled in the dialogfragment to. Nevertheless, I edited and implemented in a better way now.